### DISCUSSION TITLE
x-data incompatibility issue with htm

### DISCUSSION THREAD

**Discussion by aral** [alpinejs/alpine#3337] [7. Bugs (archived)]
Edited as per https://github.com/alpinejs/alpine/discussions/3337#discussioncomment-4397737:

[htm](https://github.com/developit/htm) converts `<div x-data … />` to `<div x-data="true" … />` whereas Alpine.js currently expects `<div x-data="" … />`.

## Suggested solution

As per https://github.com/alpinejs/alpine/discussions/3337#discussioncomment-4397836:

> I suppose you just add "or data strictly equal to true" in [here: https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-data.js#L24](https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-data.js#L24)

**Comment by SimoTod** [comment]
all the directives in alpine need x-data. I believe the documentation says that on one of the first pages.

**Reply by SimoTod** [reply]
just to clarify, x-data doesn't need to be on the element. it can be (and most of the time it will be) on one of the parent tags.

**Comment by aral** [comment]
@SimoTod Yep, it does mention that. It also mentions that you can just put `x-data` on them (incorrect) – only `x-data="{}"` works – and some of the examples leave it out completely (incorrect).

(Details/links above.)

**Reply by SimoTod** [reply]
The example you posted with just x-data works correctly for me

**Reply by aral** [reply]
Ah, odd. Then this might be something that [htm](https://github.com/developit/htm) is messing up on my end. Will attempt to reproduce using vanilla HTML and close this if I can’t.

**Reply by SimoTod** [reply]
no worries. It could be that the preprocessor is removing the attribute when it doesn't have a value. 👍 
It's a discussion so you can't really close it (this project doesn't use the issues tab).

**Reply by ekwoka** [reply]
Or more accurately, it's making it be `x-data="true"` which can be appropriate behavior for a lot of properties, but would be incorrect for x-data.

It could be worth having Alpine handle this case and not simply leave that up to the user to work around.

**Reply by SimoTod** [reply]
Yeah, it won't be too much of a change. I suppose you just add "or data strictly equal to true" in [here: https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-data.js#L24](https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-data.js#L24)

**Reply by aral** [reply]
@ekwoka @SimoTod Ah, you beat me to it. Yep, that’s what’s happening :)

**Reply by aral** [reply]
Updated the original description based on your replies.

**Reply by aral** [reply]
Pull request opened based on your suggestion, @SimoTod: https://github.com/alpinejs/alpine/pull/3339
